core/src/main/scala/org/apache/spark/scheduler/SchedulableBuilder.scala

```diff
@@ -69,60 +72,81 @@ private[spark] class FairSchedulableBuilder(val rootPool: Pool, conf: SparkConf)
   val DEFAULT_WEIGHT = 1
 
   override def buildPools() {
-    var is: Option[InputStream] = None
+    var fileData: Option[FileData] = None
     try {
-      is = Option {
-        schedulerAllocFile.map { f =>
-          new FileInputStream(f)
-        }.getOrElse {
-          Utils.getSparkClassLoader.getResourceAsStream(DEFAULT_SCHEDULER_FILE)
-        }
-      }
-
-      is.foreach { i => buildFairSchedulerPool(i) }
+      fileData = getFileData()
+      fileData.foreach { data => buildFairSchedulerPool(data) }
+    } catch {
+      case NonFatal(t) =>
+        logError("Error while building the fair scheduler pools: ", t)
+        throw t
     } finally {
-      is.foreach(_.close())
+      fileData.foreach(_.inputStream.close())
     }
 
     // finally create "default" pool
     buildDefaultPool()
   }
 
+  private def getFileData(): Option[FileData] = {
+    schedulerAllocFile.map { f =>
+      val file = new File(f)
+      val fis = new FileInputStream(file)
```

Comment:
Ah I see.  It seems like the whole path might be useful for making it super explicit which place the data is coming from though? e.g., because users often have generically defined files, like config.xml, which might exist in multiple locations.